cmd/zoekt-mirror-gitea/main.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This binary fetches all repos of a user or organization and clones
+// them.  It is strongly recommended to get a personal API token from
+// https://gitea.com/user/settings/applications, save the token in a
+// file, and point the --token option to it.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/sdk/gitea"
+
+	"github.com/sourcegraph/zoekt/gitindex"
+)
+
+type topicsFlag []string
+
+func (f *topicsFlag) String() string {
+	return strings.Join(*f, ",")
+}
+
+func (f *topicsFlag) Set(value string) error {
+	*f = append(*f, value)
+	return nil
+}
+
+type reposFilters struct {
+	topics        []string
+	excludeTopics []string
+	noArchived    *bool
+}
+
+func main() {
+	dest := flag.String("dest", "", "destination directory")
+	giteaURL := flag.String("url", "", "Gitea url. If not set gitea.com will be used as the host.")
+	org := flag.String("org", "", "organization to mirror")
+	user := flag.String("user", "", "user to mirror")
+	token := flag.String("token",
+		filepath.Join(os.Getenv("HOME"), ".gitea-token"),
+		"file holding API token.")
+	forks := flag.Bool("forks", false, "also mirror forks.")
+	deleteRepos := flag.Bool("delete", false, "delete missing repos")
+	namePattern := flag.String("name", "", "only clone repos whose name matches the given regexp.")
+	excludePattern := flag.String("exclude", "", "don't mirror repos whose names match this regexp.")
+	topics := topicsFlag{}
+	flag.Var(&topics, "topic", "only clone repos whose have one of given topics. You can add multiple topics by setting this more than once.")
+	excludeTopics := topicsFlag{}
+	flag.Var(&excludeTopics, "exclude_topic", "don't clone repos whose have one of given topics. You can add multiple topics by setting this more than once.")
+	noArchived := flag.Bool("no_archived", false, "mirror only projects that are not archived")
+
+	flag.Parse()
+
+	if *dest == "" {
+		log.Fatal("must set --dest")
+	}
+	if *giteaURL == "" && *org == "" && *user == "" {
+		log.Fatal("must set either --org or --user when gitea.com is used as host")
+	}
+
+	var host string
+	var client *gitea.Client
+	clientOptions := []gitea.ClientOption{}
+	if *giteaURL != "" {
+		*giteaURL = "https://gitea.com/"
```

Comment:
shouldn't you just make the default for the flag this?

Reply:
Good catch. Most of this file is copy/paste from the github mirror. I've cleaned this up.